src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -399,6 +409,58 @@ bool CanApplyChange(DocumentId documentId)
             }
         }
 
+        private void NotifyRefactorChanges(IEnumerable<DocumentId> changedDocs, CodeAnalysis.Solution oldSolution)
+        {
+            // Without any services to notify there's no need to dig through the documents
+            if (!_refactorNotifyServices.Any())
+            {
+                return;
+            }
+
+            foreach (var changedDocId in changedDocs)
+            {
+                var newDocument = CurrentSolution.GetDocument(changedDocId);
+
+                // Documents without syntax tree won't have the annotations attached
+                if (!newDocument.SupportsSyntaxTree)
+                {
+                    continue;
+                }
+
+                var syntaxRoot = newDocument.GetSyntaxRootSynchronously(CancellationToken.None);
+                var symbolRenameNodes = syntaxRoot.GetAnnotatedNodes(RenameSymbolAnnotation.RenameSymbolKind);
+
+                foreach (var node in symbolRenameNodes)
+                {
+                    foreach (var annotation in node.GetAnnotations(RenameSymbolAnnotation.RenameSymbolKind))
+                    {
+                        var oldDocument = oldSolution.GetDocument(changedDocId);
+
+                        if (!(newDocument.TryGetSemanticModel(out var newSemanticModel) && newSemanticModel is object))
```

Comment:
This shouldn't be using TryGetSemanticModel, since this only returns one if you still happen to have one cached. The doc comments here are hilarious:

https://github.com/dotnet/roslyn/blob/4cd280b52e1c63f57c48d405fec32f5a8cfcf553/src/Workspaces/Core/Portable/Workspace/Solution/Document.cs#L252-L253